lib/cartopy/geodesic/_geodesic.pyx

```diff
@@ -0,0 +1,240 @@
+# (C) British Crown Copyright 2015, Met Office
+#
+# This file is part of cartopy.
+#
+# cartopy is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cartopy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with cartopy.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+This module defines the core CRS class which can interface with Proj.4.
+The CRS class is the base-class for all projections defined in :mod:`cartopy.crs`.
+
+"""
+from cpython.mem cimport PyMem_Malloc
+from cpython.mem cimport PyMem_Realloc
+from cpython.mem cimport PyMem_Free
+import numpy as np
+cimport numpy as np
+from cython.parallel cimport prange
+
+cdef extern from "geodesic.h":
+    #External imports of Proj4.9 functions
+    cdef struct geod_geodesic:
+        pass
+
+    ctypedef geod_geodesic* geodesic_t
+
+    void geod_init(geodesic_t, double, double)
+    void geod_direct(geodesic_t, double, double, double, double,
+                     double*, double*, double*) nogil
+    void geod_inverse(geodesic_t, double, double, double, double,
+                      double*, double*, double*) nogil
+
+cdef class Geodesic:
+    """
+    Defines an ellipsoid on which to solve geodesic problems. 
+
+    """
+    cdef geod_geodesic* geod
+
+    def __cinit__(self, radius=6378137, flattening=1/298.257223563):
+        """
+        Create an ellipsoid with a given radius and flattening.
```

Comment:
I'm not sure this is the best description of what the class actually does is it? It should really say something about what it is used for, like solving the geodesic problem or such.